rs/utils/src/bloom_filter/immutable_bloom_filter.rs

```diff
@@ -0,0 +1,115 @@
+use std::hash::{Hash, Hasher};
+
+use anyhow::Result;
+use memmap2::Mmap;
+use xxhash_rust::xxh3::Xxh3;
+
+use crate::bloom_filter::HashIdx;
+
+pub struct ImmutableBloomFilter {
+    mmap: Mmap,
+    data_offset: usize,
+    num_blocks: usize,
+    num_hash_functions: usize,
+    block_size_in_bits: usize,
```

Comment:
let's make this a constant (512 bits). Compiler will be able to optimize much better